### ISSUE TITLE
Problems under macOS Ventura

### ISSUE THREAD

**Issue by Andreas0602** [mucommander/mucommander#784]
Problem 1: The muCommander app which I had built myself under macOS Monterey does not start after upgrading to macOS Ventura: "The app is damaged and should be moved to trash". The known fix `sudo xattr -cr /Applications/muCommander.app` does not help.

Problem 2: Building muCommander under macOS Ventura fails with the error "A problem occurred configuring project ':mucommander-core' ... BUG! exception in phase 'semantic analysis' in source unit '_BuildScript_' Unsupported class file major version 63".

The muCommander version from the hompage works fine.

**Comment by Andreas0602** [comment]
I have JDK 19.0.1 on my machine, and I use the following command to build muCommander: `git clone --recurse-submodules git@github.com:mucommander/mucommander.git && cd mucommander && ./gradlew dmg`

**Comment by ahadas** [comment]
@Andreas0602 the second issue is likely to be resolved by upgrading gradle to a version that supports Java 19, I see [here](https://docs.gradle.org/current/userguide/compatibility.html) that it's not supported yet

about the first issue, I don't have macOS Ventura yet so I can't check this but I'd try to disable gatekeeper (sudo spctl –master-disable). I don't remember it happening with a dmg that was created locally though..

**Comment by Andreas0602** [comment]
I was able to build with Java 19 under macOS Monterey without any problems. The problem started only after upgrading to Ventura.

Disabling gatekeeper sounds like opening a security hole, so I don't think I'll do that.

**Comment by ahadas** [comment]
> I was able to build with Java 19 under macOS Monterey without any problems. The problem started only after upgrading to Ventura.

that's interesting, I don't think it's related to the operating system itself but maybe you used a different version of Java 19 for which it worked. anyway, I'll track gradle to see when it supports Java 19 - it doesn't make sense to upgrade to Java 19 before that

> Disabling gatekeeper sounds like opening a security hole, so I don't think I'll do that.

right but if you do it temporarily just to install the dmg that you built and then enabling it again, it should be fine - but of course, it's your call

**Comment by Andreas0602** [comment]
I downgraded to Java 18.0.2.1. As a result, building succeeded, but macOS Ventura still complains that the app is damaged. `sudo xattr -cr /Applications/muCommander.app` does not help.

This problem did not occur under Monterey.

**Comment by Andreas0602** [comment]
The muCommander version from the hompage also has problems when copying multiple files to an external USB flash drive. Copying takes a long time, even in the case of small files, and afterwards some of the files on the USB drive cannot be opened. I tried it with two different USB flash drives. However, I'm not sure if it's a problem of muCommander. Double Commander does not even recognize the USB drives at all. On the other hand, Commander One does not have any problems with these USB drives. Strange ...

**Comment by ahadas** [comment]
@Andreas0602 thanks for reporting this. seems like we'll need to see how to  handle unsigned jar, for development purposes, on macOS Ventura and to investigate the issue you're facing with USB drivers. as I wrote elsewhere, I can't upgrade to Ventura at this time (due to restrictions on my machine) so it would need to wait a bit unless anyone else can take a look at those earlier

**Comment by Andreas0602** [comment]
Yesterday I upgraded my Mac to Ventura 13.0.1. Afterwards, macOS still complained about the muCommander app I had built by myself, but I was able to get it running by clicking "Open anyway" in System Settings>Security & Privacy.

The problems with external USB drives under Ventura seem to general, not specific to muCommander. So I guess Apple needs to fix these problems. See the follwoing threads:

https://forums.macrumors.com/threads/extremely-slow-external-ssds-and-hdds-on-macbook-pro-16-m1-max.2367355/page-2

https://www.reddit.com/r/MacOS/comments/ye92tx/usb_20_slow_transfer_speed_after_ventura_update/

I will close this issue.
